slatedb/src/map_iter.rs

```diff
@@ -30,7 +30,10 @@ impl<T: KeyValueIterator> MapIterator<T> {
 
     pub(crate) fn new_with_ttl_now(iterator: T, ttl_now: i64) -> Self {
         let f = Box::new(move |entry: RowEntry| {
-            if is_not_expired(&entry, ttl_now) {
+            // Merge operands should not be converted into tombstones and instead they
+            // will be filtered out by the merge operator (otherwise we will early terminate
+            // the merging when we should keep looking for a base value or true tombstone)
+            if is_not_expired(&entry, ttl_now) || matches!(entry.value, ValueDeletable::Merge(_)) {
```

Comment:
Remind me: are we even supporting TTL for merge? Do we (or will we) raise an error if a merge op is put'd with a TTL?

Reply:
Ah, gotcha! So this code is just passing the merge up the stack. But is it the merge operator or merge iterator that filters? The comment says the merge operator, but that would mean the user would need to do it in their merge op impl?

Reply:
ah I meant `MergeOperatorIterator` 🫠 - you can see I added that modification in this PR. I'll update the comment, but it's an unfortunate naming scheme haha.

Reply:
Haha yea, the "merge operator" part confused me.